### PR TITLE
feat: update analytics wizard

### PR DESCRIPTION
**Pull request by leogermani**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

image

This PR updates the Analytics wizard by removing the tabs handling UA's custom dimensions and events and replacing with a tab where the users can add the credentials to enable GA4 back-end events tracking.

This PR does NOT deprecates or removes several back-end classes, hooks and methods that uses those custom dimensions and other UA plus AMP event tracking. I think there should be a dedicated effort for that.

### How to test the changes in this Pull Request:

1. Go to Newspack > Analytics
2. Confirm you see only one tab other than Plugins
3. Go to that tab
4. Check that you can save your credentials and they persist
5. Try to add a Measurement ID that don't start with "G-" and confirm you see an error
6. Perform a donation or newsletter subscription in your site
7. confirm you see a [corresponding event](https://github.com/Automattic/newspack-plugin/tree/master/includes/data-events/connectors/ga4) in your Real time dashboard (it can take up to two minutes for it to show up - make sure to browse your site to trigger the async jobs)

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Comment by leogermani** [comment]
Waiting for some copy review

**Comment by leogermani** [comment]
Copy reviewed. PR is ready for code review

File: assets/wizards/analytics/views/newspack-custom-events/index.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	TextControl,
+	withWizardScreen,
+} from '../../../../components/src';
+
+/**
+ * Analytics Custom Events screen.
+ */
+class NewspackCustomEvents extends Component {
+	state = {
+		ga4Credendials: newspack_analytics_wizard_data.ga4_credentials,
```
Comment (dkoo) [inline assets/wizards/analytics/views/newspack-custom-events/index.js:26]: If you haven't set a measurement ID or secret before, the localized values will be `false`, and this appears as a stringified value in the fields.

Screen Shot 2023-07-26 at 2 22 18 PM

File: assets/wizards/analytics/views/newspack-custom-events/index.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	handleAPIError = ( { message: error } ) => this.setState( { error } );
+
+	updateGa4Credentials = () => {
+		const { wizardApiFetch } = this.props;
+		wizardApiFetch( {
+			path: '/newspack/v1/wizard/analytics/ga4-credentials',
+			method: 'POST',
+			data: {
+				measurement_id: this.state.ga4Credendials.measurement_id,
+				measurement_protocol_secret: this.state.ga4Credendials.measurement_protocol_secret,
```
Comment (dkoo) [inline assets/wizards/analytics/views/newspack-custom-events/index.js:39]: Our UI design pattern uses a "quiet" loading animation when updating a settings page (and the regular loading pattern when the entire application is loading).

```suggestion
				measurement_protocol_secret: this.state.ga4Credendials.measurement_protocol_secret,
				quiet: true,
```

File: assets/wizards/analytics/views/newspack-custom-events/index.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	Notice,
+	SectionHeader,
+	TextControl,
+	withWizardScreen,
+} from '../../../../components/src';
+
+/**
+ * Analytics Custom Events screen.
+ */
+class NewspackCustomEvents extends Component {
```
Comment (dkoo) [inline assets/wizards/analytics/views/newspack-custom-events/index.js:24]: Very much non-blocking and a stylistic choice, but worth noting that for newer components we try to prefer functional-style React components for consistency.

**Review by dkoo — CHANGES_REQUESTED** [review]
Works well, and the UI looks good! A couple of minor UI-related changes needed below.

File: assets/wizards/analytics/views/newspack-custom-events/index.js (hunk 3 above)
Comment (leogermani) [reply assets/wizards/analytics/views/newspack-custom-events/index.js:24]: yeah.. I just used the other components under Analytics as a starter and they were like this

Comment (dkoo) [reply assets/wizards/analytics/views/newspack-custom-events/index.js:24]: Yeah, those must have been old—when we have tons of free time we can refactor all of them. 😆

File: assets/wizards/analytics/views/newspack-custom-events/index.js (hunk 1 above)
Comment (leogermani) [reply assets/wizards/analytics/views/newspack-custom-events/index.js:26]: fixed in https://github.com/Automattic/newspack-plugin/pull/2580/commits/c77eaaa2442e5bb09f163b3c5a2c424fabf877f1

**Review by dkoo — APPROVED** [review]

**Pull request merged**